SConstruct

```diff
@@ -501,6 +502,14 @@ else:
     # Disable assert() for production targets (only used in thirdparty code).
     env.Append(CPPDEFINES=["NDEBUG"])
 
+env["enable_engine_debugger"] = methods.get_cmdline_bool("enable_engine_debugger", env.debug_features)
+
+if env["enable_engine_debugger"]:
+    env.Append(CPPDEFINES=["DEBUGGER_ENABLED"])
```

Comment:
I think the define should be `ENGINE_DEBUGGER_ENABLED` to match the SCons option name, but also to avoid confusion with `DEBUG_ENABLED`.